### DISCUSSION TITLE
Show launcher on startup

### DISCUSSION THREAD

**Discussion by Kcola** [wezterm/wezterm#3069] [Q&A]
Is there a way to show the launcher menu on startup?

**Comment by wez** [comment]
No. Is your goal to see the launcher and have that choose what to spawn as the initial program?

**Reply by Kcola** [reply]
yh, wanted to be able to choose between wsl or powershell on startup since I tend to use both often.

**Reply by pwsandoval** [reply]
I wanted to open a similar topic, did you get it?

**Reply by tony-sol** [reply]
Is there any solution for this?

**Comment by nekowinston** [comment]
How about this as a starting point? It seems to do what you would want:
```lua
wezterm.on("gui-startup", function(cmd)
  local tab, pane, window = wezterm.mux.spawn_window(cmd or {})
  window
    :gui_window()
    :perform_action(
      wezterm.action.ShowLauncherArgs({ flags = "DOMAINS" }),
      pane
    )
end)
```
It still creates an initial tab, but it shows the launcher once on startup.
